AirCasting/SearchAndFollow/CompleteScreen/Views/SearchAndFollowChartViewModel.swift

```diff
@@ -9,20 +9,47 @@ class SearchAndFollowChartViewModel: ObservableObject {
         let value: Double
         let color: Color
     }
+    
     struct ChartMeasurement {
         let value: Double
         let time: Date
     }
     
+    enum EntriesSensor: String {
+        case OpenAQ = "OpenAQ"
+        case AirBeam = "AirBeam"
+        case PurpleAir = "PurpleAir"
+        case undefined = ""
+    }
+    
     @Published var entries: [ChartDot] = []
     
     let numberOfEntries = 9
     
-    func generateEntries(with measurements: [ChartMeasurement], thresholds: ThresholdsValue) -> (Date?, Date?) {
+    func generateEntries(with measurements: [ChartMeasurement], thresholds: ThresholdsValue, basedOn sensorName: String) -> (Date?, Date?) {
         var times: [Date] = []
         var buffer: [ChartMeasurement] = []
+        var updatedMeasurements = measurements
+        
+        let sensor = getEntriesSensor(using: sensorName)
+        if sensor != .OpenAQ {
+            guard let firstElement = updatedMeasurements.reversed().first?.time else { return (nil, nil) }
+            
+            switch sensor {
+            case .AirBeam:
```

Comment:
I think we should be using polymorphism instead of passing `sensorName` in and doing that by switching thru possibilities in here.

In other words:
Change this code so that if we ever add another sensor type this whole file won't need to be modified at all (this is called open-closed principle and should be applied here)